catalog/services_state.go

```diff
@@ -171,8 +171,8 @@ func (state *ServicesState) ExpireServer(hostname string) {
 
 	for _, svc := range state.Servers[hostname].Services {
 		previousStatus := svc.Status
-		state.ServiceChanged(svc, previousStatus, svc.Updated)
 		svc.Tombstone()
+		state.ServiceChanged(svc, previousStatus, svc.Updated)
```

Comment:
You know it occurs to me that maybe we ought to be just doing this outside the loop by capturing the initial state and the final state. No one can process those intermediate changes that fast anyway.

Reply:
Hmm, I guess if anything relied on `svc.Updated` that would break them. Maybe there is some middle ground.